actionpack/lib/action_dispatch/http/cache.rb

```diff
@@ -167,9 +167,7 @@ def prepare_cache_control!
         end
 
         def handle_conditional_get!
-          if etag? || last_modified? || !@cache_control.empty?
```

Comment:
Looking for feedback on what negative side-effects removing all of these conditionals would have (will also see if any tests fail).